frame/vesting/src/lib.rs

```diff
@@ -374,24 +618,34 @@ impl<T: Config> Pallet<T> {
 		asset: AssetIdOf<T>,
 		schedules: Vec<VestingScheduleOf<T>>,
 	) -> DispatchResult {
-		let bounded_schedules: BoundedVec<VestingScheduleOf<T>, T::MaxVestingSchedules> =
-			schedules.try_into().map_err(|_| Error::<T>::MaxVestingSchedulesExceeded)?;
-
 		// empty vesting schedules cleanup the storage and unlock the fund
-		if bounded_schedules.len().is_zero() {
+		if schedules.len().is_zero() {
 			<VestingSchedules<T>>::remove(who, asset);
 			T::Currency::remove_lock(VESTING_LOCK_ID, asset, who)?;
 			return Ok(())
 		}
 
+		ensure!(
+			schedules.len() as u32 <= T::MaxVestingSchedules::get(),
+			Error::<T>::MaxVestingSchedulesExceeded
+		);
+
+		let bounded_schedules: BoundedBTreeMap<_, _, _> = schedules
```

Comment:
```suggestion
		let bounded_schedules = schedules
```

Reply:
Type annotation is redundant as it's also written in the `.collect()` call

Reply:
Doesn't work if I omit it (`cannot infer type`)